### PR TITLE
Fix MatchingDeclarationName false positives

### LINKED ISSUE #686: MatchingDeclarationName has false positives
In this case (filename `Whatever.kt`):

```kotlin
fun a()

private class Something
```

The MatchingDeclarationName rule flags the file — I suppose saying that it should be called `Something.kt` as it's the only class in there — but it shouldn't really as this class contains only a publicly visible top level function and that private class is only an implementation detail of the function. One such file for which this is reproducible is [this](https://github.com/squanchy-dev/squanchy-android/blob/develop/app/src/main/java/net/squanchy/tweets/service/TweetModelConverter.kt).

### PR DESCRIPTION
**Pull request by rock3r**
This PR fixes #686 in that it filters out private classes/interfaces/objects from the `MatchingDeclarationName` rule. These private classes/interfaces/objects are an implementation detail and should not determine the name of a file, whose only public member may well be a function.

In addition to the visibility filter, this PR also improves upon FindingsAssertions and uses them in the rule's negative cases tests in lieu of the more lax `assertThat(findings).hasSize(1)`.

### CODE REVIEW

**Comment by rock3r** [comment]
Good point, not sure why it's not using tabs. Will fix.

**Comment by rock3r** [comment]
Done :)

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MatchingDeclarationName.kt [hunk 1]
```diff
@@ -47,13 +48,14 @@ import org.jetbrains.kotlin.psi.KtFile
 class MatchingDeclarationName(config: Config = Config.empty) : Rule(config) {
 
 	override val issue: Issue = Issue(javaClass.simpleName, Severity.Style,
-			"If a source file contains only a single top-level class or object, " +
+			"If a source file contains only a single non-private top-level class or object, " +
```
Comment (Mauin) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MatchingDeclarationName.kt]: Can you add this non-private destinction also to the KDoc and re-generate the documentation with it? 🙃

Comment (rock3r) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MatchingDeclarationName.kt]: Sure, on it

Comment (rock3r) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MatchingDeclarationName.kt]: @Mauin Done :)

Comment (rock3r) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/MatchingDeclarationName.kt]: While verifying this I have noticed #691 happens and incomplete documentation is generated for some rules

**Comment by arturbosch** [comment]
Merged manually.

**Pull request closed without merging**